pkg/identity/identity.go

```diff
@@ -16,40 +16,63 @@ package identity
 )
 
-// OIDMatcher holds an OID field and a list of values to match on
-type OIDMatcher struct {
-	ObjectIdentifier asn1.ObjectIdentifier `yaml:"objectIdentifier"`
-	ExtensionValues  []string              `yaml:"extensionValues"`
+// CertificateIdentity holds a certificate subject and an optional list of identity issuers
+type CertificateIdentity struct {
+	CertSubject string   `yaml:"certSubject"`
+	Issuers     []string `yaml:"issuers"`
 }
 
-// CustomOID holds an OID field represented in dot notation and a list of values to match on
-type CustomExtension struct {
-	ObjectIdentifier string   `yaml:"objectIdentifier"`
-	ExtensionValues  []string `yaml:"extensionValues"`
+// MonitoredValues holds a set of values to compare against a given entry
+type MonitoredValues struct {
+	// CertificateIdentities contains a list of subjects and issuers
+	CertificateIdentities []CertificateIdentity `yaml:"certIdentities"`
+	// Fingerprints contains a list of key fingerprints. Values are as follows:
+	// For keys, certificates, and minisign, hex-encoded SHA-256 digest
+	// of the DER-encoded PKIX public key or certificate
+	// For SSH and PGP, the standard for each ecosystem:
+	// For SSH, unpadded base-64 encoded SHA-256 digest of the key
+	// For PGP, hex-encoded SHA-1 digest of a key, which can be either
+	// a primary key or subkey
+	Fingerprints []string `yaml:"fingerprints"`
+	// Subjects contains a list of subjects that are not specified in a
+	// certificate, such as a SSH key or PGP key email address
+	Subjects []string `yaml:"subjects"`
+	// OIDMatchers contains a list of OID extension fields and associated values
+	// ex. Build Signer URI, associated with specific workflow URIs
+	OIDMatchers []extensions.OIDMatcher `yaml:"oidMatchers"`
+	// FulcioExtensions contains all extensions currently supported by Fulcio
+	// each extension has a list of values to match on, ex. `build-signer-uri`
+	FulcioExtensions extensions.FulcioExtensions `yaml:"fulcioExtensions"`
+	// CustomExtensions contains a list of custom extension fields, represented in dot notation
+	// and associated values to match on.
+	CustomExtensions []extensions.CustomExtension `yaml:"customExtensions"`
 }
 
-// ParseObjectIdentifier parses a string representing an ObjectIdentifier in dot notation
-// and converts it into an asn1.ObjectIdentiifer.
-func ParseObjectIdentifier(oid string) (asn1.ObjectIdentifier, error) {
-	if len(oid) == 0 {
-		return nil, errors.New("could not parse object identifier: empty input")
-	}
-	nodes := strings.Split(oid, ".")
-	objectIdentifier := make([]int, len(nodes))
-	for i, node := range nodes {
-		if strings.TrimSpace(node) == "" {
-			return nil, errors.New("could not parse object identifier: no characters between two dots")
-		}
-		intNode, err := strconv.Atoi(node)
-		if err != nil {
-			return nil, err
+// IdentityEntry holds a certificate subject, issuer, OID extension and associated value, and log entry metadata
```

Comment:
linter prompts that this import becomes `identity.IdentityEntry`, which is redundant and proposes renaming it `Entry`; I think that removes clarity but am inclined to agree about the redundancy. I could potentially see something like `LogIdentityEntry`- is the redundancy an issue we want to deal with, and if so, does something like this work? open to any more suitable renames as well

Reply:
I tend to agree with the redundancy. What about `CertificateEntry`?

Reply:
i believe technically an entry here does not need to contain any information tied to a certificate (eg. could include an SSH/PGP key related email address under the Subjects field), so that could result in reduced clarity- If something like `RekorLogEntry` or `RekorEntry` seems reasonable, could rename it as such